### PR TITLE
Make it easy to customize image names and other defaults

### PR DESCRIPTION
**Pull request by MarSik**
This change provides a common pattern for downstream customization. Just override the _defaults.yml file and set new values for the various role defaults.

### CODE REVIEW

**Comment by MarSik** [comment]
@fromanirh @yanirq @ksimon1 Please check this out. We need a simple way to customize downstream packaging and this pattern might work.

File: playbooks/_defaults.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# This file can be used to override default
+# role values. This is useful for "downstream"
+# packaging and opinionated releases.
+---
```
Comment (fromanirh) [inline playbooks/_defaults.yml]: is not clear to me why we need both build/_defaults.yml and playbooks/_defaults.yml.

File: build/Dockerfile [hunk 2]
```diff
@@ -3,7 +3,8 @@ FROM quay.io/water-hole/ansible-operator
 COPY roles/ ${HOME}/roles/
 COPY playbooks/ ${HOME}/
 COPY watches.yaml ${HOME}/watches.yaml
+COPY _defaults.yml ${HOME}/_defaults.yml
```
Comment (fromanirh) [inline build/Dockerfile]: don't we need to use build/_defaults.yml here?

File: playbooks/_defaults.yml (hunk 1 above)
Comment (MarSik) [reply playbooks/_defaults.yml]: playbooks/_defaults.yml is empty by default and is there just to make sure the vars_files will not fail. The build one is the one supposed to replace the empty playbooks/ one with actual content.

File: build/Dockerfile (hunk 2 above)
Comment (MarSik) [reply build/Dockerfile]: You might be right. I considered the build directory to be the docker context for builds, but I am not exactly sure how the operator-sdk build works. I will test that tomorrow.

Comment (yanirq) [reply build/Dockerfile]: I believe that everything invoked from the playbook should exist in the container. 
Maybe the build folder is not the best place for it logically (by operator-sdk semantics)

File: playbooks/_defaults.yml (hunk 1 above)
Comment (yanirq) [reply playbooks/_defaults.yml]: Can you add at least as a comment what could be the variables with examples maybe ?

File: build/Dockerfile (hunk 2 above)
Comment (MarSik) [reply build/Dockerfile]: Sure, that is what the COPY does. Maybe I should just move the defaults to the root directory?

Comment (MarSik) [reply build/Dockerfile]: But we have the patch.yml and preprocessing script in build/ as well. And those are used for the same purpose (downstream customization)

Comment (yanirq) [reply build/Dockerfile]: As long as they are copied i dont see a real problem

**Pull request merged**